packages/salesforcedx-vscode-core/src/commands/templates/forceFunctionCreate.ts

```diff
@@ -22,47 +19,72 @@ import {
   SfdxCommandlet,
   SfdxWorkspaceChecker
 } from '../util';
-import { BaseTemplateCommand } from './baseTemplateCommand';
 import { FUNCTION_TYPE_JAVA, FUNCTION_TYPE_JS } from './metadataTypeConstants';
 
+import { generateFunction, Language } from '@heroku/functions-core';
+import { getRootWorkspacePath } from '../../util';
+
 const LANGUAGE_JAVA = 'java';
 const LANGUAGE_JAVASCRIPT = 'javascript';
-const LOG_NAME = 'force_function_create';
-export class ForceFunctionCreateExecutor extends BaseTemplateCommand {
+const LANGUAGE_TYPESCRIPT = 'typescript';
 
-  public build(data: FunctionInfo): Command {
-    if (data.language === LANGUAGE_JAVASCRIPT) {
-      this.metadata = FUNCTION_TYPE_JS;
-      this.setFileExtension('js');
-    } else if (data.language === LANGUAGE_JAVA) {
-      this.metadata = FUNCTION_TYPE_JAVA;
-      this.setFileExtension('java');
-    }
-    return new SfdxCommandBuilder()
-      .withDescription(nls.localize('force_function_create_text'))
-      .withArg('generate:function')
-      .withFlag('--name', data.fileName)
-      .withFlag('--language', data.language)
-      .withLogName(LOG_NAME)
-      .build();
+const LOG_NAME = 'force_function_create';
+export class ForceFunctionCreateExecutor extends LibraryCommandletExecutor<
+  any
+> {
+  constructor() {
+    super(nls.localize('force_function_create_text'), LOG_NAME, OUTPUT_CHANNEL);
   }
-
-  public runPostCommandTasks(data: FunctionInfo) {
-    const language = data.language;
+  public async run(response: ContinueResponse<FunctionInfo>): Promise<boolean> {
+    const { fileName, language } = response.data;
+    let metadata: MetadataInfo | undefined;
+    switch (language) {
+      case LANGUAGE_JAVASCRIPT:
+        metadata = MetadataDictionary.getInfo(FUNCTION_TYPE_JS);
+        metadata!.suffix = '.js';
+        this.telemetry.addProperty('language', 'node');
+        break;
+      case LANGUAGE_TYPESCRIPT:
+        metadata = MetadataDictionary.getInfo(FUNCTION_TYPE_JS);
+        metadata!.suffix = '.ts';
+        this.telemetry.addProperty('language', 'node');
+        break;
+      case LANGUAGE_JAVA:
+        metadata = MetadataDictionary.getInfo(FUNCTION_TYPE_JAVA);
+        metadata!.suffix = '.java';
+        this.telemetry.addProperty('language', 'java');
+        break;
+    }
+    const { path: functionPath, welcomeText } = await generateFunction(
+      fileName,
+      language as Language,
+      getRootWorkspacePath()
+    );
+    channelService.appendLine(
+      `Created ${language} function ${fileName} in ${functionPath}.`
+    );
+    if (welcomeText) channelService.appendLine(welcomeText);
+    channelService.showChannelOutput();
+    const outputFile = metadata!.pathStrategy.getPathToSource(
+      functionPath,
+      fileName,
+      metadata!.suffix
+    );
+    const document = await vscode.workspace.openTextDocument(outputFile);
+    vscode.window.showTextDocument(document);
+    channelService.appendLine('Installing dependencies...');
```

Comment:
I wonder if it's also worth using `spawn` instead of `exec` below and streaming the install output so users can see specifically what is going on

Reply:
In this particular case, product wanted this to be a background process with limited user visibility and we don't show any notification popups either.

Reply:
Ah I seem should I remove this line as well then?

Reply:
As discussed on slack, this can remain as is.